lib/src/native/realm_core.dart

```diff
@@ -1413,36 +1429,37 @@ class _RealmCore {
       _realmLib.realm_app_config_set_platform_version(handle._pointer, Platform.operatingSystemVersion.toCharPtr(arena));
 
       _realmLib.realm_app_config_set_sdk(handle._pointer, 'Dart'.toCharPtr(arena));
-      _realmLib.realm_app_config_set_sdk_version(handle._pointer, ''.toCharPtr(arena));
+      _realmLib.realm_app_config_set_sdk_version(handle._pointer, libraryVersion.toCharPtr(arena));
 
+      //TODO: set cpu architecture when Core merges https://github.com/realm/realm-core/pull/6256#pullrequestreview-1275624284
       _realmLib.realm_app_config_set_cpu_arch(handle._pointer, ''.toCharPtr(arena));
 
-      _realmLib.realm_app_config_set_device_name(handle._pointer, ''.toCharPtr(arena));
-      _realmLib.realm_app_config_set_device_version(handle._pointer, ''.toCharPtr(arena));
+      final deviceName = getDeviceName();
+      _realmLib.realm_app_config_set_device_name(handle._pointer, deviceName.toCharPtr(arena));
+      
+      final deviceVersion = getDeviceVersion();
+      _realmLib.realm_app_config_set_device_version(handle._pointer, deviceVersion.toCharPtr(arena));
```

Comment:
According to me deviceName and deviceVerison could be members of AppConfigurationInternal.

Reply:
Internal members are needed only if accessed on multiple places within the sdk libraries ie (configuration.dart, set.dart, realm_object.dart etc)